lib/rates/FeeProvider.ts

```diff
@@ -242,7 +242,9 @@ class FeeProvider {
       typeof percentageType === 'number'
         ? percentageType
         : percentageType[orderSide],
-      referral?.premium(pair, type),
+      type === SwapType.Chain
+        ? referral?.premium(pair, type, orderSide)
+        : referral?.premium(pair, type),
```

Comment:
```suggestion
      referral?.premium(pair, type, orderSide)
```
The `orderSide` is ignored when it's not needed